compiler/luci/pass/src/FuseHorizontalFCLayers.cpp

```diff
@@ -0,0 +1,192 @@
+    return false;
+
+  if (not check_type_and_shape_equality(left_fc_weights, right_fc_weights))
+    return false;
+
+  // Lets check left and right FC bias: type and shape
+  auto left_fc_bias = dynamic_cast<luci::CircleConst *>(left_fc_node->bias());
+  auto right_fc_bias = dynamic_cast<luci::CircleConst *>(right_fc_node->bias());
+
+  // Support only if both biases are const, or both are non-const
+  // TODO Support the case that one FC has a const bias and another FC has no bias.
+  if ((left_fc_bias == nullptr and right_fc_bias != nullptr) or
+      (left_fc_bias != nullptr and right_fc_bias == nullptr))
+  {
+    return false;
+  }
+
+  // Both left/right bias are const. Check dtype/shape.
+  if (left_fc_bias != nullptr and not check_type_and_shape_equality(left_fc_bias, right_fc_bias))
+    return false;
+
+  // Both left/right bias are non-const. Check left/right fc has no bias.
+  if (left_fc_bias == nullptr)
+  {
+    auto left_no_bias = dynamic_cast<luci::CircleOutputExclude *>(left_fc_node->bias());
+    auto right_no_bias = dynamic_cast<luci::CircleOutputExclude *>(right_fc_node->bias());
+    if (not left_no_bias or not right_no_bias)
+      return false;
+  }
+
+  if (left_fc_weights->dtype() != loco::DataType::FLOAT32)
+    return false;
+
+  if (left_fc_bias != nullptr)
+  {
+    if (left_fc_bias->dtype() != loco::DataType::FLOAT32)
+      return false;
+  }
+
+  // Lets create fused FC weights and bias
+  auto fused_fc_weights = luci::clone(left_fc_weights);
+  luci::add_origin(fused_fc_weights, luci::composite_origin({luci::get_origin(left_fc_weights),
+                                                             luci::get_origin(right_fc_weights)}));
+
+  luci::CircleConst *fused_fc_bias = nullptr;
+  if (left_fc_bias != nullptr)
+  {
+    fused_fc_bias = luci::clone(left_fc_bias);
+    luci::add_origin(fused_fc_bias, luci::composite_origin({luci::get_origin(left_fc_bias),
+                                                            luci::get_origin(right_fc_bias)}));
+  }
+
+  assert(fused_fc_weights->dtype() == loco::DataType::FLOAT32);
+  sum_const_values<loco::DataType::FLOAT32>(fused_fc_weights, right_fc_weights);
+
+  if (fused_fc_bias != nullptr)
+  {
+    assert(fused_fc_bias->dtype() == loco::DataType::FLOAT32);
+    sum_const_values<loco::DataType::FLOAT32>(fused_fc_bias, right_fc_bias);
+  }
+
+  // Create fused FC node
+  auto graph = left_fc_node->graph();
+  auto fused_fc_node = graph->nodes()->create<luci::CircleFullyConnected>();
+  fused_fc_node->input(left_fc_node->input());
+  fused_fc_node->weights(fused_fc_weights);
+  if (fused_fc_bias)
+  {
+    fused_fc_node->bias(fused_fc_bias);
+  }
+  else
+  {
+    assert(nullptr !=
+           dynamic_cast<luci::CircleOutputExclude *>(left_fc_node->bias())); // FIX ME UNLESS
+    fused_fc_node->bias(left_fc_node->bias());
+  }
+
+  fused_fc_node->fusedActivationFunction(left_fc_node->fusedActivationFunction());
```

Comment:
You should use add_nodes's activation.

Reply:
Oh, I misunderstood, now I added fusedActivationFunction from `add_node`